pkg/applicationserver/io/packages/redis/registry.go

```diff
@@ -215,16 +216,15 @@ func (r ApplicationPackagesRegistry) Set(ctx context.Context, ids ttnpb.Applicat
 					return errInvalidIdentifiers
 				}
 			} else {
-				if err := ttnpb.ProhibitFields(sets,
-					"ids.end_device_ids.application_ids",
-					"ids.end_device_ids.device_id",
-					"ids.f_port",
-				); err != nil {
-					return errInvalidFieldmask.WithCause(err)
-				}
 				if err := cmd.ScanProto(updated); err != nil {
 					return err
 				}
+				if ttnpb.HasAnyField(sets, "ids.end_device_ids") && !pb.EndDeviceIdentifiers.Equal(updated.EndDeviceIdentifiers) {
```

Comment:
This incorrect `ttnpb.HasAnyField(sets, "ids.end_device_ids")` returns `false` if e.g. `"ids.end_device_ids.application_ids"` is specified.

Try for yourself:
```
diff --git a/pkg/ttnpb/fieldmask_utils_test.go b/pkg/ttnpb/fieldmask_utils_test.go
index 887f61a5c..b448d670e 100644
--- a/pkg/ttnpb/fieldmask_utils_test.go
+++ b/pkg/ttnpb/fieldmask_utils_test.go
@@ -94,6 +94,7 @@ func TestHasAnyField(t *testing.T) {
 		"a",
 		"b.c",
 		"d",
+		"e.f.g",
 	}
 	a.So(ttnpb.HasAnyField(requested, "x", "a"), should.BeTrue)
 	a.So(ttnpb.HasAnyField(requested, "x.y", "b"), should.BeFalse)
@@ -101,6 +102,7 @@ func TestHasAnyField(t *testing.T) {
 	a.So(ttnpb.HasAnyField(requested, "x", "b.c.d"), should.BeTrue)
 	a.So(ttnpb.HasAnyField(requested, "d"), should.BeTrue)
 	a.So(ttnpb.HasAnyField(requested, "d.e", "b"), should.BeTrue)
+	a.So(ttnpb.HasAnyField(requested, "e.f"), should.BeTrue)
 }
 
 func TestFlattenPaths(t *testing.T) {
```

Reply:
`pb.EndDeviceIdentifiers.Equal` uses reflection *and* we don't care about anything else besides appID/devID.
Why not just use these fields explicitly both in paths and in comparison?